src/components/HeartRateZoneGraph.vue

```diff
@@ -357,23 +359,27 @@ export default {
       let totalSeconds = 0
 
       // Process each data point and calculate heart rate zone and total time
+      // TODO optimize calculation
       sessions.forEach((session) => {
         if (session.records == null) return
+        console.time('totalSteps')
         for (let i = 0; i < session.records.length - 1; i++) {
           const entry = session.records[i]
-          const nextEntry = session.records[i + 1]
+          if (entry == null || entry.heartRate == null) continue
 
-          if (entry.heartRate == null || nextEntry.heartRate == null) continue
+          let { nextEntry, nextIndex } = this.getNextValidEntry(session, entry, i)
+          i = nextIndex // skip loop to latest valid entry
```

Comment:
if `i = nextIndex`, next loop `i` will be `+1`  since `i++`

```
records[0] -> entry
records[1] -> nextEntry
records[2] -> next loop entry will be this  ?
```

Reply:
0-1
1-2
2-2

Reply:
Ah, it should be `i = nextIndex -1`